feat: add support for applying existing customization specs (#2053)
hashicorp/terraform-provider-vsphere: website/docs/r/guest_os_customization.html.markdown

---
subcategory: "Virtual Machine"
layout: "vsphere"
page_title: "VMware vSphere: vsphere_guest_os_customization"
sidebar_current: "docs-vsphere-data-guest-os-customization"
description: |-
  Provides a VMware vSphere guest customization spec data source. This can be used to apply the customization spec when virtual machine is cloned
---

# vsphere\_guest\_os\_customization

> The `vsphere_guest_os_customization`resource can be used to create Guest OS Customizations for VM

Reviewer: ```suggestion
The `vsphere_guest_os_customization` resource can be used to a customization specification for a guest operating system.
```

~> **NOTE:** The name attribute is unique identifier for the guest OS spec per VC.

## Example Usage

```hcl
  resource "vsphere_guest_os_customization" "windows_customization" {

>     name = "win-1"

Reviewer: ```suggestion
    name = "windows-spec"
```

    type = "Windows"
    spec {
        windows_options {
            run_once_command_list = ["command-1", "command-2"]

>             computer_name = "Windows-Computer-Name"

Reviewer: ```suggestion
            computer_name = "windows"
```

            auto_logon = false
            auto_logon_count = 0

>             admin_password = "Pass-123"

Reviewer: ```suggestion
            admin_password = "VMware1!"
```

>             time_zone = 125 #(UTC+02:00) Helsinki, Kyiv, Riga, Sofia, Tallinn, Vilnius

Reviewer: ```suggestion
            time_zone = 004
```

>             workgroup = "Windows-Workgroup"

Reviewer: ```suggestion
            workgroup = "workgroup"
```

        }
    }
  }
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required) The name of the guest os customization. It is unique identifier for the guest OS customization per VC

Reviewer: ```suggestion
* `name` - (Required) The name of the customization specification is the unique identifier per vCenter Server instance.
```

> * `type` - (Optional) The type of the guest OS. Currently Linux and Windows are supported.

Reviewer: ```suggestion
* `type` - (Required) The type of customization specification: One among: Windows, Linux.
```

> * `description` - (Optional)The description of the Guest OS Customization spec.

Reviewer: ```suggestion
* `description` - (Optional) The description for the customization specification..
```

* `spec` - Container object for the Guest OS properties about to be customized . See [virtual machine customizations](#virtual-machine-customizations)

## Attribute Reference

> * `last_update_time` - The time of last modification of the Guest OS Customization spec.

Reviewer: ```suggestion
* `last_update_time` - The time of last modification to the customization specification.
```

> * `change_version` - The number of last change version of the Guest OS Customization spec.

Reviewer: ```suggestion
* `change_version` - The number of last changed version to the customization specification.
```
